pkg/ccl/backupccl/backup_planning.go

```diff
@@ -1602,6 +1379,280 @@ func checkForNewTables(
 	return nil
 }
 
+func getBackupDetailAndManifest(
```

Comment:
Since we're anyways moving things around, I'd love if this method could also be refactored to callout to helper methods instead of doing so much inline. We can leave that for a follow-up but just something to consider.